frappe/utils/backups.py

```diff
@@ -636,6 +671,61 @@ def backup(
 		"backup_path_private_files": odb.backup_path_private_files,
 	}
 
+@frappe.whitelist()
+def get_backup_encryption_key():
+	if 'backup_encryption_key' in frappe.local.conf:
+		message = frappe.local.conf.backup_encryption_key
+	else:
+		message = "No key found."
+	return frappe.msgprint(message,'Backup Encryption Key')
+
+def backup_decryption(file_path,passphrase):
+	"""
+	Decrypts backup at the given path using the passphrase.
+	"""
+	if os.path.exists(file_path):
+
+		os.rename(file_path, file_path + ".gpg")
+		file_path = file_path + ".gpg"
+		
+		cmd_string = ("gpg --yes --passphrase {passphrase} --pinentry-mode loopback -o {decryptedfile} -d {filelocation}")
+		command = cmd_string.format(
+			passphrase = passphrase,
+			filelocation = file_path,
+			decryptedfile = file_path[:-4],
+		)
+
+	frappe.utils.execute_in_shell(command)
+
+
+def backup_encryption_key():
+	"""
+	Checks if backup encryption key exists
+		else create one
+	Return:
+		Backup encryption key 
+	"""
+	from frappe.installer import update_site_config
+	if 'backup_encryption_key' not in frappe.local.conf:
+		confirm = click.confirm(
+			"No encrytion key found. Generate new Key?"
+		)
+		if not confirm:
+			sys.exit(1)
+		backup_encryption_key = Fernet.generate_key().decode()
+		update_site_config('backup_encryption_key', backup_encryption_key)
+		frappe.local.conf.backup_encryption_key = backup_encryption_key
+	return frappe.local.conf.backup_encryption_key
+	
+def decryption_rollback(file_path):
+	"""
+	Rollback if the decrypted file exist.
```

Comment:
Too ambiguous. What does rollback mean in this context? This docstring needs re-wording.

Reply:
It can be revert decryption? The idea is to reverse the decryption every time an error occurs or if the process is completed.